test/zzz-correct-index-names.sh

```diff
@@ -24,25 +24,59 @@ es_pod=$( get_es_pod es )
 es_ops_pod=$( get_es_pod es-ops )
 es_ops_pod=${es_ops_pod:-$es_pod}
 
-# INFRA_PROJECTS=$(oc set env ds/logging-fluentd --list | grep OCP_OPERATIONS_PROJECTS | sed "s/.*=//")
-# for project in ${INFRA_PROJECTS} ; do
+OPS_NAMESPACES="default openshift openshift-infra openshift-this-is-a-test"
 
-# Testing to revert the change in pr/900
-# Since logs under the tag openshift-* (e.g., openshift-web-console) are indexed in .operations,
-# this test should not fail by those logs.
-for project in default openshift openshift-infra ; do
+# write some logs from namespace openshift and openshift-infra
+test_template=$OS_O_A_L_DIR/hack/testing/templates/test-template.yaml
+
+for project in $OPS_NAMESPACES ; do
+    delete_project=""
+    if oc get project $project 2>&1 | artifact_out ; then
+        os::log::debug "use existing project $project"
+    else
+        os::log::info Creating project $project
+        oc adm new-project $project --node-selector='' 2>&1 | artifact_out
+        os::cmd::try_until_success "oc get project $project" 2>&1 | artifact_out
+        delete_project="$project"
+    fi
+    message_uuid=$( uuidgen | sed 's/[-]//g' )
+    oc process -f $test_template \
+        -p TEST_POD_NAME=test-pod \
+        -p TEST_POD_MESSAGE="$message_uuid" \
+        -p TEST_POD_SLEEP_TIME=1 \
+        -p TEST_NAMESPACE_NAME=${project} \
+        -p TEST_ITERATIONS=1 | oc create -f - 2>&1 | artifact_out
+    os::cmd::try_until_text "oc get -n ${project} pods test-pod" "^test-pod.* Running "
+    # The query part will return more than one if successful - due to the fuzzy matching,
+    # it may return results from more than one namespace - the jq select will ensure that
+    # the namespace name matches exactly
+    os::cmd::try_until_text "curl_es $es_ops_pod /.operations.*/_search?q=message:$message_uuid | \
+        jq '.hits.hits | map(select(._source.kubernetes.namespace_name == \"${project}\")) | length | . > 0'" "^true\$"
+    oc delete -n ${project} --force pod test-pod 2>&1 | artifact_out
+    os::cmd::try_until_failure "oc get -n ${project} pod test-pod"
+    if [ -n "$delete_project" ] ; then
+        oc delete project $delete_project 2>&1 | artifact_out
+        os::cmd::try_until_failure "oc get project $delete_project" 2>&1 | artifact_out
+    fi
+done
+
+for project in $OPS_NAMESPACES ; do
     qs='{"query":{"term":{"kubernetes.namespace_name":"'"${project}"'"}}}'
-    os::cmd::expect_success_and_not_text "curl_es $es_pod /_cat/indices" "project\.${project}\."
+    os::cmd::expect_success_and_not_text "curl_es $es_pod /_cat/indices" "project.${project}."
```

Comment:
do we not want to escape the regex here and make sure its an actual '.' vs a regex '.' ?